stix2/STIXPatternVisitor.py

```diff
@@ -0,0 +1,353 @@
+import importlib
+import inspect
+
+import six
+from stix2patterns.grammars.STIXPatternLexer import STIXPatternLexer
+from stix2patterns.grammars.STIXPatternParser import (STIXPatternParser,
+                                                      TerminalNode)
+from stix2patterns.grammars.STIXPatternVisitor import STIXPatternVisitor
+from stix2patterns.validator import STIXPatternErrorListener
+
+from antlr4 import CommonTokenStream, InputStream
+from stix2.patterns import _BooleanExpression
+
+# need to import all classes because we need to access them via globals()
+from .patterns import (BinaryConstant, BooleanConstant, FloatConstant,
+                       FollowedByObservationExpression, HexConstant,
+                       IntegerConstant, RepeatQualifier, StartStopQualifier,
+                       StringConstant, TimestampConstant, WithinQualifier)
+from .patterns import *  # noqa
+
+
+def collapse_lists(lists):
+    result = []
+    for c in lists:
+        if isinstance(c, list):
+            result.extend(c)
+        else:
+            result.append(c)
+    return result
+
+
+def values_only(things):
```

Comment:
This method could use better variable names and a better function name in order to be more descriptive.

Reply:
changed to:

```
def remove_terminal_nodes(parse_tree_nodes):
    values = []
    for x in parse_tree_nodes:
        if not isinstance(x, TerminalNode):
            values.append(x)
    return values
```